editor/test/integration/editor_extensions_test.clj

```diff
@@ -375,20 +377,158 @@
+      (is (thrown? LuaError (coerce coerce/null 12))))
+
+    (testing "boolean"
+      (is (true? (coerce coerce/boolean true)))
+      (is (false? (coerce coerce/boolean false)))
+      (is (thrown? LuaError (coerce coerce/boolean nil)))
+      (is (thrown? LuaError (coerce coerce/boolean "asd"))))
+
+    (testing "to-boolean"
+      (is (true? (coerce coerce/to-boolean true)))
+      (is (false? (coerce coerce/to-boolean false)))
+      (is (false? (coerce coerce/to-boolean nil)))
+      (is (true? (coerce coerce/to-boolean "asd"))))
+
+    (testing "any"
+      (is (= {:a 1} (coerce coerce/any {:a 1})))
+      (is (= "foo" (coerce coerce/any "foo")))
+      (is (= [1 2 3] (coerce coerce/any [1 2 3])))
+      (is (= false (coerce coerce/any false))))
+
+    (testing "userdata"
+      (is (= :key (coerce coerce/userdata (vm/wrap-userdata :key))))
+      (is (thrown? LuaError (coerce coerce/userdata "key")))
+      (is (= inc (coerce coerce/userdata inc))))
+
+    (testing "function"
+      (let [f (vm/invoke-1 vm (vm/bind (vm/read "return function() end" "test.lua") vm))]
+        (is (= f (coerce coerce/function f)))
+        ;; Clojure's functions are wrapped in userdata
+        (is (thrown? LuaError (coerce coerce/function inc)))))
+
+    (testing "wrap-with-pred"
+      (let [with-pred (coerce/wrap-with-pred coerce/integer even? "should be even")]
+        (is (= 2 (coerce with-pred 2)))
+        (is (thrown? LuaError (coerce with-pred 1)))
+        (is (thrown? LuaError (coerce with-pred "not an int")))))
+
+    (testing "coll-of"
+      (is (= [] (coerce (coerce/coll-of coerce/string) {:a 1}))) ;; non-integer keys are ignored
+      (is (= [] (coerce (coerce/coll-of coerce/string) {})))
+      (is (= ["a"] (coerce (coerce/coll-of coerce/string) {1 "a" :key "val"})))
+      (is (= ["a"] (coerce (coerce/coll-of coerce/string) {1 "a" -1 "b" 0 "c"}))) ;; non-positive keys are ignored
+      (is (= [1 2 3] (coerce (coerce/coll-of coerce/integer) [1 2 3])))
+      (is (= [1 2 3 nil 5] (coerce (coerce/coll-of coerce/any) [1 2 3 nil 5])))
+      (is (= [1 2 3 nil 5] (coerce (coerce/coll-of (coerce/one-of coerce/integer coerce/null)) [1 2 3 nil 5])))
+      (is (thrown? LuaError (coerce (coerce/coll-of coerce/integer) [1 2 3 nil 5])))
+      (is (= [nil nil nil nil 5] (coerce (coerce/coll-of (coerce/one-of coerce/integer coerce/null)) {5 5})))
+      (is (thrown? LuaError (coerce (coerce/coll-of coerce/integer) {5 5})))
+      (testing "min-count"
+        (is (= [1 2] (coerce (coerce/coll-of coerce/any :min-count 2) [1 2])))
+        (is (= [1 2 3] (coerce (coerce/coll-of coerce/any :min-count 2) [1 2 3])))
+        (is (thrown? LuaError (coerce (coerce/coll-of coerce/any :min-count 2) [1]))))
+      (testing "distinct"
+        (is (= [1 2 3] (coerce (coerce/coll-of coerce/any :distinct true) [1 2 3])))
+        (is (thrown? LuaError (coerce (coerce/coll-of coerce/any :distinct true) [1 2 1])))))
```

Comment:
Is `coll-of` the best name if we're strictly coercing to vectors?

Reply:
Yes, I'll rename to `vector-of`